File: Math challenge/Challanges/Addition.m
//
//  Addition.m
//  Math challenge
//
//  Created by Lukas Bischof on 09.01.18.
//  Copyright © 2018 Lukas Bischof. All rights reserved.
//

#import "Addition.h"

@implementation Addition {
    NSInteger _level;
}

+ (instancetype)newWithLevel:(NSInteger)level
{
    return [[Addition alloc] initWithLevel:level];
}

- (instancetype)initWithLevel:(NSInteger)level
{
    if ((self = [super init])) {
        _level = level;
    }
    
    return self;
}

- (Challange *)next {
    NSInteger lowerBound = (NSInteger)powf(10.f, (float)_level);
    NSInteger upperBound = (NSInteger)powf(10.f, (float)(_level + 1)) - 1;
    
    NSInteger first = randr(lowerBound, upperBound);
    NSInteger second = randr(lowerBound, upperBound);
    
    return [[Challange alloc] initWithText:SWF(@"%ld + %ld", (long)first, (long)second) andAnswer:first + second];
}

- (NSString *)challangeDescription
{
    return SWF(@"Addition level %ld", (long)_level);
}

@end

File: Math challenge/challenges/Addition.swift
//
// Created by Lukas Bischof on 24.10.19.
// Copyright (c) 2019 Lukas Bischof. All rights reserved.
//

import Foundation

class Addition: MathChallengeGenerator {
  let lowerBound: Int
  let upperBound: Int
  let level: Int
  var challengeDescription: String {
    get {
      return "Addition \(level + 1)-stellig"
    }
  }

  init(level: Int) {
    self.level = level
    self.lowerBound = Int(powf(10.0, Float(level)))
    self.upperBound = Int(powf(10.0, Float(level + 1))) - 1
  }

  func next() -> Challenge {
    let first = randomInt(lowerBound, upperBound)
    let second = randomInt(lowerBound, upperBound)

    return Challenge(text: "\(first) + \(second)", andAnswer: first + second)
  }
}
